tapfreighter/wallet.go

```diff
@@ -1654,61 +1654,31 @@ func addAnchorPsbtInputs(btcPkt *psbt.Packet, vPkt *tappsbt.VPacket,
 	}
 
 	// Now that we've added an extra input, we'll want to re-calculate the
-	// total weight of the transaction, so we can ensure we're paying
-	// enough in fees.
-	var (
-		weightEstimator     input.TxWeightEstimator
-		inputAmt, outputAmt int64
-	)
-	for _, pIn := range btcPkt.Inputs {
-		inputAmt += pIn.WitnessUtxo.Value
-
-		inputPkScript := pIn.WitnessUtxo.PkScript
-		switch {
-		case txscript.IsPayToWitnessPubKeyHash(inputPkScript):
-			weightEstimator.AddP2WKHInput()
-
-		case txscript.IsPayToScriptHash(inputPkScript):
-			weightEstimator.AddNestedP2WKHInput()
-
-		case txscript.IsPayToTaproot(inputPkScript):
-			weightEstimator.AddTaprootKeySpendInput(
-				txscript.SigHashDefault,
-			)
-		default:
-			return fmt.Errorf("unknown pkScript: %x",
-				inputPkScript)
+	// total vsize of the transaction and the necessary fee at the desired
+	// fee rate.
+	inputScripts := make([][]byte, 0, len(btcPkt.Inputs))
+	for inputIdx, input := range btcPkt.Inputs {
+		// TODO(jhb): Support non-witness UTXOs as inputs by fetching
+		// their PkScripts here.
+		if input.WitnessUtxo == nil {
+			return fmt.Errorf("input %d has no witness, not "+
+				"supported", inputIdx)
 		}
-	}
-	for _, txOut := range btcPkt.UnsignedTx.TxOut {
-		outputAmt += txOut.Value
 
-		addrType, _, _, err := txscript.ExtractPkScriptAddrs(
-			txOut.PkScript, params,
-		)
-		if err != nil {
-			return err
+		if input.WitnessUtxo.PkScript == nil {
```

Comment:
Ah, and I guess we should do a `len() == 0` check here instead of nil.

Reply:
Wish there was a linter for this!